crates/holochain/src/conductor/conductor.rs

```diff
@@ -1186,6 +1188,121 @@ pub async fn integration_dump(vault: &EnvRead) -> ConductorApiResult<Integration
         .await
 }
 
+/// Dump the full integration json state.
+/// Careful! This will return a lot of data.
+pub async fn full_integration_dump(
+    vault: &EnvRead,
+    dht_ops_cursor: Option<u64>,
+) -> ConductorApiResult<FullIntegrationStateDump> {
+    vault
+        .async_reader(move |txn| {
+            let integrated = query_dht_ops_from_statement(
+                &txn,
+                "
+                    SELECT 
+                        Header.blob as header_blob,
+                        Entry.blob as entry_blob,
+                        DhtOp.type as dht_type,
+                        DhtOp.hash as dht_hash,
+                        DhtOp.rowid as rowid
+                    FROM Header
+                    JOIN
+                        DhtOp ON DhtOp.header_hash = Header.hash
+                    LEFT JOIN
+                        Entry ON Header.entry_hash = Entry.hash
+                    WHERE when_integrated IS NOT NULL
+                ",
+                dht_ops_cursor,
+            )?;
+
+            let validation_limbo = query_dht_ops_from_statement(
+                &txn,
+                "
+                    SELECT 
+                        Header.blob as header_blob,
+                        Entry.blob as entry_blob,
+                        DhtOp.type as dht_type,
+                        DhtOp.hash as dht_hash,
+                        DhtOp.rowid as rowid
+                    FROM Header
+                    JOIN
+                        DhtOp ON DhtOp.header_hash = Header.hash
+                    LEFT JOIN
+                        Entry ON Header.entry_hash = Entry.hash
+                    WHERE when_integrated IS NULL
+                    AND (
+                        (is_authored = 1 AND validation_stage IS NOT NULL AND validation_stage < 3)
+                        OR
+                        (is_authored = 0 AND (validation_stage IS NULL OR validation_stage < 3))
+                    )
+                ",
+                dht_ops_cursor,
+            )?;
+
+            let integration_limbo = query_dht_ops_from_statement(
+                &txn,
+                "
+                    SELECT 
+                        Header.blob as header_blob,
+                        Entry.blob as entry_blob,
+                        DhtOp.type as dht_type,
+                        DhtOp.hash as dht_hash,
+                        DhtOp.rowid as rowid
+                    FROM Header
+                    JOIN
+                        DhtOp ON DhtOp.header_hash = Header.hash
+                    LEFT JOIN
+                        Entry ON Header.entry_hash = Entry.hash
+                    WHERE when_integrated IS NULL AND validation_stage = 3
+                ",
+                dht_ops_cursor,
+            )?;
```

Comment:
Seems like it would be more efficient to combine these three queries and do the filtering in memory, but this is fine too.